src/ProtocolGateway.IotHubClient/IotHubClient.cs

```diff
@@ -80,7 +80,7 @@ public class IotHubClient : IMessagingServiceClient
             {
                 var identity = (IotHubDeviceIdentity)deviceIdentity;
                 csb.AuthenticationMethod = DeriveAuthenticationMethod(csb.AuthenticationMethod, identity);
-                csb.HostName = identity.IotHubHostName;
+                //csb.HostName = identity.IotHubHostName;
```

Comment:
Client connection can be set to Protocol gateway which has a different hostname than the IoT Hub. For example it connects to localhost as the Protocol gateway and Protocol Gateway connects to cloud. This line was overriding the cloud hostname with localhost.